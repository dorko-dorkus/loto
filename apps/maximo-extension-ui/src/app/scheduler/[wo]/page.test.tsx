import { fireEvent, render, screen } from '@testing-library/react';
import { vi, test, expect } from 'vitest';
import Page from './page';

test('renders virtualized gantt and updates conflicts', async () => {
  class ResizeObserver {
    observe() {}
    unobserve() {}
    disconnect() {}
  }
  // @ts-ignore
  global.ResizeObserver = ResizeObserver;

  vi.stubEnv('NEXT_PUBLIC_USE_API', 'true');
  const apiBase = process.env.NEXT_PUBLIC_API_BASE ?? '';

  const sample = {
    schedule: [
      {
        date: '2024-01-01',
        p10: 10,
        p50: 20,
        p90: 30,
        price: 40,
        hats: 1,
        conflicts: ['Conflict 0']
      },
      {
        date: '2024-01-02',
        p10: 12,
        p50: 22,
        p90: 32,
        price: 42,
        hats: 2,
        conflicts: []
      }
    ],
    seed: 'abc',
    objective: 0.95,
    blocked_by_parts: false,
    rulepack_sha256: 'abc123'
  };
  const fetchMock = vi
    .spyOn(global, 'fetch')
    .mockImplementation((input: RequestInfo) => {
      if (typeof input === 'string' && input === apiBase + '/schedule') {
        return Promise.resolve({ ok: true, json: async () => sample } as Response);
      }
      if (typeof input === 'string' && input === apiBase + '/hats') {
        return Promise.resolve({ ok: true, json: async () => [] } as Response);
      }
      return Promise.resolve({ ok: true, json: async () => ({}) } as Response);
    });

  render(<Page params={{ wo: 'WO-1' }} />);
  const rows = await screen.findAllByTestId('gantt-row-date');
  fireEvent.click(rows[0]);
  expect(screen.getByLabelText('Conflict 0')).toBeInTheDocument();
  fetchMock.mockRestore();
});

