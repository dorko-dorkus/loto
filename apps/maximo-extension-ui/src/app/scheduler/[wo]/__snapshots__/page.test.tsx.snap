// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders gantt, price curve and hats timeline 1`] = `
<main
  class="h-full flex flex-col"
>
  <h1
    class="mb-4 text-xl font-semibold"
  >
    Scheduler: 
    WO-1
  </h1>
  <div
    class="w-full h-full"
  >
    <div
      data-sync="schedule"
      data-testid="gantt-chart"
    >
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 200px; min-width: 0;"
      />
    </div>
    <div
      data-sync="schedule"
      data-testid="price-chart"
    >
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 100px; min-width: 0;"
      />
    </div>
    <div
      data-sync="schedule"
      data-testid="hats-chart"
    >
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 100px; min-width: 0;"
      />
    </div>
  </div>
  <footer
    class="mt-4 text-sm text-gray-500"
    data-testid="schedule-meta"
  >
    Seed: 
    abc
     Objective: 
    0.95
  </footer>
</main>
`;
